offer-manager-app/src/main/groovy/life/qbic/portal/offermanager/components/product/copy/CopyProductViewModel.groovy

```diff
@@ -0,0 +1,76 @@
+import life.qbic.datamodel.dtos.business.ProductId
+import life.qbic.datamodel.dtos.business.services.DataStorage
+import life.qbic.datamodel.dtos.business.services.MetabolomicAnalysis
+import life.qbic.datamodel.dtos.business.services.PrimaryAnalysis
+import life.qbic.datamodel.dtos.business.services.Product
+import life.qbic.datamodel.dtos.business.services.ProjectManagement
+import life.qbic.datamodel.dtos.business.services.ProteomicAnalysis
+import life.qbic.datamodel.dtos.business.services.SecondaryAnalysis
+import life.qbic.datamodel.dtos.business.services.Sequencing
+import life.qbic.datamodel.dtos.general.Person
+import life.qbic.portal.offermanager.communication.EventEmitter
+import life.qbic.portal.offermanager.components.offer.create.ProductItemViewModel
+import life.qbic.portal.offermanager.components.product.MaintainProductsViewModel
+import life.qbic.portal.offermanager.components.product.create.CreateProductViewModel
+import life.qbic.portal.offermanager.dataresources.persons.CustomerResourceService
+import life.qbic.portal.offermanager.dataresources.persons.ProjectManagerResourceService
+import life.qbic.portal.offermanager.dataresources.products.ProductsResourcesService
+
+
+/**
+ * <h1>Holds all values that the user specifies in the CreateProductView</h1>
+ *
+ * @since 1.0.0
+ *
+*/
+class CopyProductViewModel extends CreateProductViewModel{
+
+    EventEmitter<Product> productUpdate
+    ProductId productId
+    CopyProductViewModel(EventEmitter<Product> productUpdate) {
+        super()
+        this.productUpdate = productUpdate
+        this.productUpdate.register((Product product) -> {
+            loadData(product)
+        })
+    }
+
+    private void loadData(Product product) {
+        productName = product.productName
+        productDescription = product.description
+        productUnit = product.unit
+        productUnitPrice = product.unitPrice
+        setProductCategory(product)
+        productId = product.productId
+    }
+
+    private ProductCategory setProductCategory(Product product){
+        switch(product.class) {
+            case DataStorage:
+                productCategory = ProductCategory.DATA_STORAGE
+                break
+            case PrimaryAnalysis:
+                productCategory = ProductCategory.PRIMARY_BIOINFO
+                break
+            case ProjectManagement:
+                productCategory = ProductCategory.PROJECT_MANAGEMENT
+                break
+            case SecondaryAnalysis:
+                productCategory = ProductCategory.SECONDARY_BIOINFO
+                break
+            case Sequencing:
+                productCategory = ProductCategory.SEQUENCING
+                break
+            case ProteomicAnalysis:
+                productCategory = ProductCategory.PROTEOMIC
+                break
+            case  MetabolomicAnalysis:
+                productCategory = ProductCategory.METABOLOMIC
+                break
+        }
```

Comment:
Why do you need it?

Reply:
Because the productcategory is not stored in the product itself but has to be determined by the productclass

Reply:
you could use the Converter to do that. That's something that needs to be done at a lot of places so we can reduce redundancy :)

Reply:
But isn't the converter domain logic and should be seperated from the app logic? Furthermore it only provides method to generate a product or determine the category from product classes. 
Although i agree that we should probably define a helper class that's accessible everywhere and stores this conversion since it's done a lot in our current code base

Reply:
I'll move it to the Converter for this PR and update it in the PR that introduces the converter interface